irodsConnector/session.py

```diff
@@ -15,28 +15,64 @@
 
 class Session(object):
     """Irods session operations """
-    _session = None
-    context = utils.context.Context()
 
-    def __init__(self, password=''):
+    def __init__(self, context: utils.context.Context, password=None, greet=False):
         """ iRODS authentication with Python client.
 
         Parameters
         ----------
         password : str
             Plain text password.
+        context : utils.context Context object containing valid
+            irods environment and ibridges configuration 
+        greet: bool
+            Print some greeting and irods info to prompt
 
-        The 'password' property can autoload from its cache, but can be
+        The 'password' is autoloaded from its cache, but can be
         overridden by `password` argument.  The iRODS authentication
         file is expected in the standard location (~/.irods/.irodsA) or
         to be specified in the local environment with the
         IRODS_AUTHENTICATION_FILE variable.
 
         """
         self._password = password
+        self._ienv = context.irods_environment.config
+        self._conf = context.ibridges_configuration.config
+        self._irods_env_file = context.irods_env_file
+        self._irods_session = None
```

Comment:
```suggestion
        self.irods_session = None
```
As far as I understand the irods_session should be a public attribute, so no underscore then.